subsys/shell/shell.c

```diff
@@ -1314,6 +1320,10 @@ int shell_init(const struct shell *shell, const void *transport_config,
 {
 	__ASSERT_NO_MSG(shell);
 	__ASSERT_NO_MSG(shell->ctx && shell->iface && shell->default_prompt);
+	BUILD_ASSERT(SHELL_THREAD_PRIORITY >=
+		          K_HIGHEST_APPLICATION_THREAD_PRIO
+		        && SHELL_THREAD_PRIORITY <= K_LOWEST_APPLICATION_THREAD_PRIO,
+		        "Invalid range for thread priority");
```

Comment:
Suggestion: FYI, `BUILD_ASSERT` does not need to be placed inside a function. It would be nice to have this right below `SHELL_THREAD_PRIORITY` definition so that it is readily accessible when looking up `SHELL_THREAD_PRIORITY`.

Reply:
Ack. implemented the change.